modules/ztp-image-based-upgrade-generate-seed-image.adoc

```diff
@@ -0,0 +1,296 @@
+// Module included in the following assemblies:
+// Epic TELCOSTRAT-160 (4.15/4.16), story TELCODOCS-1576
+// * scalability_and_performance/ztp-talm-updating-managed-policies.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ztp-image-based-upgrade-seed-generation_{context}"]
+= Generating a seed image with the {lcao}
+
+Use the {lcao} to generate the seed image with the `SeedGenerator` CR. The {lcao} will do the following:
+
+* Checks for required system configurations
+* Performs any necessary system cleanup before generating the seed image
+* Launches the image generation tool, which does the following:
```

Comment:
>I think it is important to notice two things here as well:

Did you mean to say "note" instead of "notice"?

Reply:
correct, thx

Reply:
>I think it is important to note two things here as well:
The image generation tool or task calls recert (not sure if it is going to be productized) which is in charge of expiring all the crypto information stored in the seed cluster. cc/ @omertuc
That's important so private crypto information of a running/valid SNO (the seed cluster) cannot be obtained by pulling the seed container image. It can be the case that could obtain the kubeconfig /credentials to log into the seed cluster.

That's a bit of an implementation detail, I don't want users to know what recert is, or even think about their crypto, so I'm against including that in the doc text

Reply:
I agree, the idea is to give @amolnar-rh more detail. However, I think it is important to let the users know that the cryptographic certificates are expired in the seed and regenerated again in the target. The users do not have to worry about the container seed image being obtained by a non-trusted person and then it can extract or obtain credentials/keys of the seed and connect or add some malicious code.

Reply:
>The users do not have to worry

Sure, if we want to communicate that assurance then we could have something like that then.

Just one nit: 

> the cryptographic certificates are expired

Technically expiring the certificates is not the reason the user shouldn't have to worry, the thing that recert does that makes it safe is replacing all the private keys / JWTs with different random ones. Also let's say that this happens but without mentioning recert, just that it happens

Reply:
Okay, so it's clear that the recert tool shouldn't be mentioned, but it's a good idea to say something about the recertification process happening. I currently have an overview of the process, which mentions it as:
> The LCA will do the following:
> [...]
> * Launches the image generation task, which involves the following:
> ** Stops the cluster operators
> ** Prepares the seed image configuration
> ** Generates and pushes the seed image to the image repository specified in the `imageBasedUpgrade` CR
> ** Restores the cluster operators
> ** Expires seed cluster certifications
> * Generates new certificates for the seed cluster
> [...]

I have a few references to the recert tool, I'll remove them in the next commit.